### DISCUSSION TITLE
Provide a new way an attirubes are initialized

### DISCUSSION THREAD

**Discussion by feelingnothing** [DSharpPlus/DSharpPlus#989] [Ideas]
Due to limitations of the constructor parameters type i think it will be suitable to the library to initialize attributes with static function like `Initialize` and pass services to this function, this would give more space for all kind of custom attributes that people are creating and giving a database access to the attributes is definetly a plus for commands

**Comment by Naamloos** [comment]
You can pass dependencies to the CommandsNext framework. In the `ExecuteCheckAsync(CommandContext ctx..` method for your Check attribute, you can use the `CommandContext` to get your `IServiceCollection`. From there, you can grab your dependency (e.g. your database).
